### PR TITLE
Update docker builder image dockerfile to allow data prepper build

### LINKED ISSUE #1106: [RELEASE] Data Prepper 1.2.0 to Docker
### Is your feature request related to a problem? Please describe

The Data Prepper team plans to release Data Prepper to Docker.

Planned release date: Dec 2, 2021.

### Describe the solution you'd like

We discussed this with @peterzhuamazon. Based on this discussion we plan to create a single script which will build Data Prepper, build the Docker image, and push to Docker Hub. The `docker-build` Jenkins job will be able to run this script with the necessary permissions to perform this push.

### Describe alternatives you've considered

_No response_

### Additional context

The Data Prepper work to create this change is:
https://github.com/opensearch-project/data-prepper/issues/623

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Update docker builder image dockerfile to allow data prepper build.
Include awscli 1.22 as well as jdk14.
Thanks.

 
### Issues Resolved
#1106 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dlvenable — APPROVED** [review]

File: docker/ci/dockerfiles/ci-runner.docker-builder.ubuntu2004.x64.dockerfile [hunk 1]
```diff
@@ -17,7 +17,17 @@ FROM ubuntu:20.04
 RUN apt-get update -y && apt-get install -y software-properties-common && add-apt-repository ppa:jacob/virtualisation -y
 
 # Install necessary packages
-RUN apt-get update -y && apt-get upgrade -y && apt-get install -y binfmt-support qemu qemu-user qemu-user-static docker.io curl && apt clean -y
+RUN apt-get update -y && apt-get upgrade -y && apt-get install -y binfmt-support qemu qemu-user qemu-user-static docker.io curl python3-pip && apt clean -y && pip3 install awscli==1.22.12
+
+# Install JDK
+RUN curl -SL https://github.com/AdoptOpenJDK/openjdk14-binaries/releases/download/jdk-14.0.2%2B12/OpenJDK14U-jdk_x64_linux_hotspot_14.0.2_12.tar.gz -o /opt/jdk14.tar.gz && \
```
Comment (gaiksaya) [inline docker/ci/dockerfiles/ci-runner.docker-builder.ubuntu2004.x64.dockerfile:23]: we didn't need this to build opensearch and opensearch dashboards docker images?

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/ci-runner.docker-builder.ubuntu2004.x64.dockerfile:23]: Like I said in the description it is for data-prepper.

Comment (gaiksaya) [reply docker/ci/dockerfiles/ci-runner.docker-builder.ubuntu2004.x64.dockerfile:23]: I think I didnt ask it properly. Let me try again:

How were we able to build opensearch and opensearch-dashboards images without jdk

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/ci-runner.docker-builder.ubuntu2004.x64.dockerfile:23]: Because in opensearch and dashboards we use shell scripts.
Data prepper is using gradle to build.

Comment (gaiksaya) [reply docker/ci/dockerfiles/ci-runner.docker-builder.ubuntu2004.x64.dockerfile:23]: I see! Thanks!

**Review by gaiksaya — APPROVED** [review]

**Review by abhinavGupta16 — APPROVED** [review]

**Comment by dblock** [comment]
Curious why we aren't using https://github.com/opensearch-project/opensearch-build/blob/main/docker/ci/dockerfiles/ci-runner.centos7.dockerfile for this?

**Comment by peterzhuamazon** [comment]
> Curious why we aren't using https://github.com/opensearch-project/opensearch-build/blob/main/docker/ci/dockerfiles/ci-runner.centos7.dockerfile for this?

This can only be done for x64 ubuntu specifically.
